### PR TITLE
Feature/backend auth

### PR DESCRIPTION
**Pull request by HaicoDorenbos**

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -183,6 +183,7 @@ const (
 // Handshake represents the handshake message payload.
 type Handshake struct {
 	Version semver.Version `json:"version"`
+	Token string `json:"token"`
```
Comment (golangci) [inline pkg/api/api.go]: File is not goimports-ed

File: pkg/trcapi/conn.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ type decoder interface {
 // Conn is safe for concurrent use by multiple goroutines.
 type Conn struct {
 	version semver.Version
+	Token string
```
Comment (golangci) [inline pkg/trcapi/conn.go]: File is not goimports-ed

File: pkg/api/api.go [hunk 3]
```diff
@@ -183,6 +183,7 @@ const (
 // Handshake represents the handshake message payload.
 type Handshake struct {
 	Version semver.Version `json:"version"`
+	Token 	string 				 `json:"token"`
```
Comment (golangci) [inline pkg/api/api.go]: File is not goimports-ed

File: pkg/webapi/webapi.go [hunk 4]
```diff
@@ -38,6 +36,12 @@ func StateHandler(pool *trcapi.Pool) http.HandlerFunc {
 			return
 		}
 
+		//Check if the authentication token matches
+		if trcConn.Token != r.Header.Get("token") {
+			http.Error(w, fmt.Sprintf("Wrong authentication token"), http.StatusUnauthorized)
+			trcConn.Close()
```
Comment (rvolosatovs) [inline pkg/webapi/webapi.go]: Connection should not be closed here, also token should be checked ASAP before doing anything else, we want to fail early.(applies to other endpoints as well)

File: pkg/trcapi/conn.go [hunk 5]
```diff
@@ -96,14 +97,17 @@ func Connect(ver semver.Version, w io.Writer, r io.Reader) (*Conn, error) {
 
 	resp := &api.Handshake{
 		Version: hs.Version,
+		Token:   hs.Token,
```
Comment (rvolosatovs) [inline pkg/trcapi/conn.go]: The token should be decoded from the message, which arrives *from* TRC. We don't send the token to the TRC.

**Pull request merged**